scripts/ai-sdlc/generate-spec.mjs

```
@@ -25,23 +26,57 @@ const slug = ISSUE_TITLE
 const padded = String(ISSUE_NUMBER).padStart(4, '0');
 const specFile = `docs/specs/${padded}-${slug}.md`;
 
+// Build a source-file tree so the agent can reference accurate paths and
+// spot which files exist before writing the spec. Capped at 300 entries to
+// keep prompt size bounded.
+function scanDir(dir, depth = 0, acc = []) {
+  if (depth > 3 || acc.length > 300) return acc;
```

ESLint

[error] Expected { after 'if' condition.

(curly)